firmware/robot2015/cpu/ball_sense.hpp

```diff
@@ -0,0 +1,38 @@
+// Determines if the emitter to reciever beam is broken
+//  while accounting for ambiant light
+#include "mbed.h"
+
+class ball_sense {
+public:
+    ball_sense();
+
+    // Call once per cycle
+    void update_ball_sensor();
+
+    // Returns true if beam is broken (Ball is contained)
+    bool have_ball();
+
+private:
+    DigitalOut* emitter_pin;
```

Comment:
Let's make the DigitalOut and AnalogIn regular variables, rather than pointers.  It's slightly safer and faster